File: app/src/main/java/com/mecong/tenderalarm/sleep_assistant/media_selection/NoisesFragment.java
package com.mecong.tenderalarm.sleep_assistant.media_selection;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.mecong.tenderalarm.R;
import com.mecong.tenderalarm.sleep_assistant.SleepAssistantPlayListModel;

public class NoisesFragment extends Fragment implements NoisesItemViewAdapter.NoisesItemClickListener {
    private static final String SELECTED_POSITION = "selectedPosition";
    private int selectedPosition;
    private NoisesItemViewAdapter adapter;
    private SleepAssistantPlayListModel model;


    private NoisesFragment(SleepAssistantPlayListModel model) {
        this.model = model;
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param selectedPosition - selected item in the list
     * @return A new instance of fragment NoisesFragment.
     */
    static NoisesFragment newInstance(int selectedPosition, SleepAssistantPlayListModel model) {
        NoisesFragment fragment = new NoisesFragment(model);
        Bundle args = new Bundle();
        args.putInt(SELECTED_POSITION, selectedPosition);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (getArguments() != null) {
            selectedPosition = getArguments().getInt(SELECTED_POSITION);
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_noises, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {

        final RecyclerView recyclerView = view.findViewById(R.id.noisesList);

        recyclerView.setLayoutManager(new LinearLayoutManager(view.getContext()));
        adapter = new NoisesItemViewAdapter(view.getContext(),
                SleepNoise.Companion.retrieveNoises(), selectedPosition);
        adapter.setClickListener(this);

        recyclerView.setAdapter(adapter);
    }

    @Override
    public void onItemClick(View view, int position) {
        SleepAssistantPlayListModel.PlayList newPlayList = new SleepAssistantPlayListModel.PlayList(
                adapter.getItem(position).getUrl(), adapter.getItem(position).getName(), SleepMediaType.NOISE);
        model.setPlaylist(newPlayList);
    }
}

File: app/src/main/java/com/mecong/tenderalarm/sleep_assistant/media_selection/NoisesFragment.kt
package com.mecong.tenderalarm.sleep_assistant.media_selection

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.mecong.tenderalarm.databinding.FragmentNoisesBinding
import com.mecong.tenderalarm.model.PropertyName
import com.mecong.tenderalarm.model.SQLiteDBHelper
import com.mecong.tenderalarm.sleep_assistant.Media
import com.mecong.tenderalarm.sleep_assistant.SleepAssistantPlayListActive
import com.mecong.tenderalarm.sleep_assistant.SleepAssistantPlayListIdle
import com.mecong.tenderalarm.sleep_assistant.media_selection.SleepNoise.Companion.retrieveNoises
import org.greenrobot.eventbus.EventBus

class NoisesFragment : Fragment(), NoisesItemClickListener {
  private var selectedPosition = 0
  private lateinit var adapter: NoisesItemViewAdapter
  private lateinit var noises: List<Media>

  private var _binding: FragmentNoisesBinding? = null

  // This property is only valid between onCreateView and onDestroyView.
  private val binding get() = _binding!!

  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
    _binding = FragmentNoisesBinding.inflate(inflater, container, false)
    return binding.root
  }

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    binding.noisesList.layoutManager = LinearLayoutManager(view.context)

    val sqLiteDBHelper = SQLiteDBHelper.sqLiteDBHelper(this.requireContext())!!
    noises = retrieveNoises(this.requireContext())

    val savedActiveTab = sqLiteDBHelper.getPropertyInt(PropertyName.ACTIVE_TAB)

    selectedPosition = -1
    if (savedActiveTab == 2) {
      selectedPosition = sqLiteDBHelper.getPropertyInt(PropertyName.TRACK_NUMBER) ?: 0
      initPlaylist(selectedPosition, false)
    }

    adapter = NoisesItemViewAdapter(this.context, noises, selectedPosition)
    adapter.setClickListener(this)
    binding.noisesList.adapter = adapter
    binding.noisesList.scrollToPosition(if (selectedPosition > -1) selectedPosition else 0)
  }

  override fun onItemClick(view: View?, position: Int) {
    initPlaylist(position, true)
  }

  private fun initPlaylist(position: Int, active: Boolean) {
    val newPlayList = if (active)
      SleepAssistantPlayListActive(position, noises, SleepMediaType.NOISE, -1)
    else
      SleepAssistantPlayListIdle(position, noises, SleepMediaType.NOISE, -1)

    EventBus.getDefault().postSticky(newPlayList)
  }
}